draft-ietf-mls-protocol.md

```diff
@@ -1098,16 +1098,35 @@ an asymmetric key pair with some associated data:
 
 * A public key (for the HPKE scheme in use, see {{ciphersuites}})
 * A private key (only within the member's direct path, see {{views}})
-* A credential (only for leaf nodes)
-* An ordered list of "unmerged" leaves (see {{views}})
 * A hash of certain information about the node's parent, as of the last time the
   node was changed (see {{parent-hashes}}).
 
+Moreover, leaf nodes contains:
+* A credential and signature verification key
+* A source, telling how the leaf node was introduced in the tree (see
+  {{leaf-node-contents}})
+* An epoch at which it was added (if the source is a key package) or at
+  which it was last updated (if the source is an update or commit).
+
 Every node, regardless of whether the node is blank or populated, has
 a corresponding _hash_ that summarizes the contents of the subtree
 below that node.  The rules for computing these hashes are described
 in {{tree-hashes}}.
 
+Nodes also have a corresponding _last update epoch_, which corresponds to the
+epoch in which their content was last modified.  For non-blank leaves, this last
+update epoch is stored in their content (if their source is an update or commit).
+For non-blank nodes, the last update epoch is the last epoch at which one of its
+leaves was updated, in other words it is the maximum last update epoch of its
+leaves.
+
+A leaf can be added under a node's subtree after this node was last updated,
+in that case we say that this leaf is _unmerged_ for this node.  More precisely,
+a leaf is unmerged for a node iff. its source is key package and its add epoch
+is greater or equal than the last update epoch of the node. The list of leaves
+that are unmerged for a node is called the list of _unmerged leaves_ for this
+node.
```

Comment:
Nit: We don't use "iff" elsewhere, and it shouldn't be followed by a "."

Reply:
This would be clearer with something like: a leaf node L is unmerged with with regard to a parent node P if:

* P is an ancestor of L
* `L.leaf_node_source == key_package`
* `L.add_epoch > P.last_update_epoch`